class.jetpack.php

```diff
@@ -4713,6 +4718,70 @@ public static function build_authorize_url( $redirect = false, $iframe = false )
 		return add_query_arg( $args, $api_url );
 	}
 
+	/**
+	 * This action fires at the beginning of the Manager::authorize method.
+	 */
+	public static function authorize_starting() {
+		$jetpack_unique_connection = Jetpack_Options::get_option( 'unique_connection' );
+		// Checking if site has been active/connected previously before recording unique connection.
+		if ( ! $jetpack_unique_connection ) {
+			// jetpack_unique_connection option has never been set.
+			$jetpack_unique_connection = array(
+				'connected'    => 0,
+				'disconnected' => 0,
+				'version'      => '3.6.1',
+			);
+
+			update_option( 'jetpack_unique_connection', $jetpack_unique_connection );
+
+			// Track unique connection.
+			$jetpack = self::init();
+
+			$jetpack->stat( 'connections', 'unique-connection' );
+			$jetpack->do_stats( 'server_side' );
+		}
+
+		// Increment number of times connected.
+		$jetpack_unique_connection['connected'] += 1;
+		Jetpack_Options::update_option( 'unique_connection', $jetpack_unique_connection );
+	}
+
+	/**
+	 * This action fires at the end of the Manager::authorize method when a secondary user is
+	 * linked.
+	 */
+	public static function authorize_ending_linked() {
+		// Don't activate anything since we are just connecting a user.
+		self::state( 'message', 'linked' );
+	}
+
+	/**
+	 * This action fires at the end of the Manager::authorize method when the master user is
+	 * authorized.
+	 *
+	 * @param array $data The request data.
+	 */
+	public static function authorize_ending_authorized( $data ) {
+		// If this site has been through the Jetpack Onboarding flow, delete the onboarding token.
+		self::invalidate_onboarding_token();
+
+		// If redirect_uri is SSO, ensure SSO module is enabled.
+		parse_str( wp_parse_url( $data['redirect_uri'], PHP_URL_QUERY ), $redirect_options );
+
+		/** This filter is documented in class.jetpack-cli.php */
+		$jetpack_start_enable_sso = apply_filters( 'jetpack_start_enable_sso', true );
+
+		$activate_sso = (
+			isset( $redirect_options['action'] ) &&
+			'jetpack-sso' === $redirect_options['action'] &&
+			$jetpack_start_enable_sso
+		);
+
+		$do_redirect_on_error = ( 'client' === $data['auth_type'] );
+
+		self::handle_post_authorization_actions( $activate_sso, $do_redirect_on_error );
```

Comment:
I'm not sure if everything in [handle_post_authorization_actions()](https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L6945) should stay in the Jetpack class. Specifically, I'm looking into whether these calls need to be moved in Manager::authorize():

* Jetpack_Options::delete_raw_option( 'jetpack_last_connect_url_check' );
* wp_clear_scheduled_hook( 'jetpack_clean_nonces' );
		wp_schedule_event( time(), 'hourly', 'jetpack_clean_nonces' );

Reply:
As discussed in Slack, we agree that moving them to areas of respective concerns in separate PRs